plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeWriter.java

```diff
@@ -187,7 +188,8 @@ public DataFileInfo getDataFileInfo()
     {
         TrinoInputFile inputFile = fileSystem.newInputFile(rootTableLocation.appendPath(relativeFilePath));
         Map<String, Type> dataColumnTypes = columnHandles.stream()
-                .collect(toImmutableMap(DeltaLakeColumnHandle::getBasePhysicalColumnName, DeltaLakeColumnHandle::getBasePhysicalType));
+                // Lowercase because the subsequent logic expects lowercase
```

Comment:
`Map<String, Type>` -> `Map</* lowercase */ String, Type>`

same in `readStatistics` & `mergeStats` params

.. or switch to `CanonicalColumnName` as map keys

Reply:
Can we switch back to the original names here instead of doing it remotely in `io.trino.plugin.deltalake.DeltaLakeMetadata#appendAddFileEntries` ?